057-deprecate-and-remove-envvar-config-provider.md

```diff
@@ -0,0 +1,47 @@
+# Deprecate and remove EnvVarConfigProvider
+
+This proposes deprecating the [kafka-env-var-config-provider](https://github.com/strimzi/kafka-env-var-config-provider) project. It also proposes a timeline to archive the project and stop including the JAR in Strimzi releases.
+
+## Current situation
+
+In [proposal 30](https://github.com/strimzi/proposals/blob/main/030-env-var-config-provider.md), we added an implementation of the [ConfigProvider](https://kafka.apache.org/35/javadoc/org/apache/kafka/common/config/provider/ConfigProvider.html) interface of Apache Kafka that allows retrieving configuration values at runtime from environment variables. This implementation lives in the `kafka-env-var-config-provider` project. This is especially useful to safely configure sensitive settings in virtualized/containerized environments like Kubernetes.
+
+For example a Kafka client can use this provider with:
+```properties
+config.providers=env
+config.providers.env.class=io.strimzi.kafka.EnvVarConfigProvider
+option1=${env::FIRST_ENV_VAR}
+```
+
+## Motivation
+
+Since Apache Kafka 3.5.0, there's a built-in `ConfigProvider` implementation that works with environment variables. It was added via [KIP-887](https://cwiki.apache.org/confluence/display/KAFKA/KIP-887%3A+Add+ConfigProvider+to+make+use+of+environment+variables). It provides the same functionality as Strimzi's implementation.
+
+Example usage:
+```properties
+config.providers=env
+config.providers.env.class=org.apache.kafka.common.config.provider.EnvVarConfigProvider
+option1=${env:FIRST_ENV_VAR}
+```
+
+## Proposal
+
+Once Strimzi only supports Kafka versions >= 3.5.0, we should:
+- deprecate the [kafka-env-var-config-provider](https://github.com/strimzi/kafka-env-var-config-provider) project
+- update all references to `io.strimzi.kafka.EnvVarConfigProvider` by `org.apache.kafka.common.config.provider.EnvVarConfigProvider`. This includes code, examples, documentation, etc
+
+After 4 Strimzi releases have happened or Strimzi reached 1.0.0 (whichever comes first), we will archive the `kafka-env-var-config-provider` project and stop including the JAR in Strimzi releases.
```

Comment:
There is anyway effort for releasing a new version when we make updates for dependencies or? It's always more work.
Anyway not strong opinion from me.

Reply:
It does not have any real dependencies. The last release we did was just around Java 8 / Java 17.

Reply:
I struggle to imagine what kind of incompatibility would cause it to stop working. That would be a regression in Apache Kafka itself. In any case, I don't think this wording would be helpful to users. Even though it's deprecated we should still support it until it's removed. It's important that deprecation timelines are something users can reason about.

Reply:
I agree with Tom that clear timelines are important for users so they know when they need to take actions.

Reply:
@scholzj Does that answer your concern?